Services/UI/OJS.Services.Ui.Business/Implementations/ContestCategoriesBusinessService.cs

```diff
@@ -75,6 +76,29 @@ public async Task<IEnumerable<ContestCategoryListViewModel>> GetAllParentCategor
         return categories;
     }
 
+    public bool IsCategoryChildOfInvisibleParentRecursive(int? categoryId)
+    {
+        if (categoryId == null)
+        {
+            return false;
+        }
+
+        var categoryWithParent = this.contestCategoriesData
+            .GetByIdQuery(categoryId.Value).Include(c => c.Parent).FirstOrDefault();
+
+        if (categoryWithParent?.Parent != null)
+        {
+            if (categoryWithParent.Parent.IsVisible == false)
+            {
+                return true;
+            }
+
+            return this.IsCategoryChildOfInvisibleParentRecursive(categoryWithParent.Parent.ParentId);
```

Comment:
Don't you have to provide the `.Parent.Id` here, as with `ParentId` you will skip the Parent's check?